File: src/main/java/com/JamieSatchell/ims/domain/Stock.java
package com.JamieSatchell.ims.domain;

import jakarta.persistence.*;
import lombok.NoArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Stock {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "location_id")
    private Location location;
    private Integer quantity;
    @ManyToOne(cascade = CascadeType.MERGE)
    private Product product;
    public void setLocationTEST(Location locationItem){
        this.location=locationItem;
        locationItem.addStock(this);
    }
}

File: src/main/java/com/JamieSatchell/ims/domain/Stock.kt
package com.JamieSatchell.ims.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToOne
import jakarta.persistence.ManyToMany
import lombok.NoArgsConstructor



@Entity
@NoArgsConstructor
data class Stock (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,
        @OneToOne
        var location: Location? = null,
        var quanity: Int? = null,
        @ManyToMany
        var products: MutableList<Product> = mutableListOf()

)
